File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    queue<int> temp;
    
    
    for(int i = 0; i < arr.size(); i++){
        if(i == arr.size() - 1 || arr[i] != arr[i + 1]){
            temp.push(arr[i]);
        }
    }
    
   
    while(!temp.empty()){
        answer.push_back(temp.front());
        temp.pop();
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int count = 0;
        Queue<Integer> q = new LinkedList<>();
        q.add(arr[0]); 
        for(int i=1;i<arr.length;i++){
           if(arr[i-1]==arr[i]){
               continue;
           }
            q.add(arr[i]);    

        }
        int[] answer = new int[q.size()];
        int size = q.size();
        for(int i = 0;i<size;i++){
            answer[i] = q.poll();
        }
        return answer;
    }
}